src/librustc_typeck/diagnostics.rs

```diff
@@ -4665,6 +4625,22 @@ It is recommended that you look for a `new` function or equivalent in the
 crate's documentation.
 "##,
 
+E0642: r##"
+This error indicates that there is a mismatch between generic parameters and
+impl Trait parameters in a trait Declaration versus it's impl.
+
+```compile_fail,E0642
+#![feature(conservative_impl_trait)]
+trait Foo
+    fn foo(&self, &impl Iterator)
```

Comment:
[Unnamed parameters are deprecated](https://github.com/rust-lang/rfcs/pull/1685). Consider changing this to `fn foo(&self, iter: &impl Iterator)`

Reply:
I just noticed that I need to change the feature as well.

Reply:
@cramertj It is sufficient to go from `&impl Iterator` to `_: &impl Iterator`, correct?

Reply:
Sure, I suppose you could do that.